h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/predict/XGBoostNativeVariableImportance.java

```diff
@@ -1,26 +1,53 @@
 package hex.tree.xgboost.predict;
 
-import hex.tree.xgboost.XGBoostModelInfo;
 import hex.tree.xgboost.util.BoosterHelper;
 import hex.tree.xgboost.util.FeatureScore;
 import ml.dmlc.xgboost4j.java.Booster;
 import ml.dmlc.xgboost4j.java.XGBoostError;
+import org.apache.log4j.Logger;
+import water.Key;
 
 import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
 import java.util.HashMap;
 import java.util.Map;
 
 public class XGBoostNativeVariableImportance implements XGBoostVariableImportance {
-    
-    private final byte[] boosterBytes;
+
+    private static final Logger LOG = Logger.getLogger(XGBoostNativeVariableImportance.class);
+
     private final File featureMapFile;
 
-    public XGBoostNativeVariableImportance(XGBoostModelInfo modelInfo, File featureMapFile) {
-        this.boosterBytes = modelInfo._boosterBytes;
-        this.featureMapFile = featureMapFile;
+    public XGBoostNativeVariableImportance(Key modelKey, String featureMap) {
+        featureMapFile = createFeatureMapFile(modelKey, featureMap);
+    }
+
+    private File createFeatureMapFile(Key modelKey, String featureMap) {
+        try {
+            File fmFile = Files.createTempFile("h2o_xgb_" + modelKey.toString(), ".txt").toFile();
+            fmFile.deleteOnExit();
```

Comment:
The JVM process could potentially exit somewhere in next century. Scoping these temporary files with the model building process SOMEHOW would be cool (reactive/event driven ?) - not a showstopper right now. This is general need in H2O I'd say.

Reply:
the deleteOnExit is just a precaution, normally the file gets deleted during cleanup()